gitlab/mixins.py

```diff
@@ -312,6 +312,7 @@ class UpdateMixin(_RestManagerBase):
     _parent_attrs: Dict[str, Any]
     _path: Optional[str]
     _update_uses_post: bool = False
+    _update_uses_patch: bool = False
```

Comment:
Should we use an `enum` or something for this?

Or check that only one is set to `True`?

Reply:
I put together a PR here: https://github.com/python-gitlab/python-gitlab/pull/2685

If you like it you can merge it or incorporate it into this PR.

Reply:
Makes sense, but would be cool if we could use that, or whatever we go with, to get rid of the helper method and its if/elif/else clauses. Haven't thought about it too much though.